infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/fsm/VerifyingRemoteSinkLeaderState.java

```diff
@@ -77,8 +78,19 @@ public void onEntry(LogReplicationRuntimeState from) {
         log.debug("onEntry :: Verifying Remote Leader, transition from {}", from.getType());
         log.trace("Submitted tasks to worker :: size={} activeCount={} taskCount={}", worker.getQueue().size(),
                 worker.getActiveCount(), worker.getTaskCount());
-        // Verify Leadership on connected nodes (ignore those for which leadership is pending)
-        this.worker.submit(this::verifyLeadership);
+
+        // Proceed if remoteLeader is known, otherwise verify Leadership on connected nodes (ignore those for which
```

Comment:
what does it mean by 'ignore those for which...'.  Should it be 'ignore those... connection is not up'?

Reply:
This isn't a good comment. I have changed it to say:

> Proceed if remoteLeader is known. Only if local is connection starter for the session and the remoteLeader is not known, verify Leadership on connected nodes

I probably meant to say the second statement above.